ios/Video/RCTVideo.swift

```diff
@@ -1331,73 +1334,79 @@ class RCTVideo: UIView, RCTVideoPlayerViewControllerDelegate, RCTPlayerObserverH
+                                       "canPlayFastForward": NSNumber(value: _playerItem.canPlayFastForward),
+                                       "canPlaySlowForward": NSNumber(value: _playerItem.canPlaySlowForward),
+                                       "canPlaySlowReverse": NSNumber(value: _playerItem.canPlaySlowReverse),
+                                       "canStepBackward": NSNumber(value: _playerItem.canStepBackward),
+                                       "canStepForward": NSNumber(value: _playerItem.canStepForward),
+                                       "naturalSize": [
+                                           "width": width,
+                                           "height": height,
+                                           "orientation": orientation,
+                                       ],
+                                       "audioTracks": audioTracks,
+                                       "textTracks": self._textTracks?.compactMap { $0.json } ?? textTracks.map(\.json),
+                                       "target": self.reactTag as Any])
+                }
 
-            if self._videoLoadStarted {
-                let audioTracks = await RCTVideoUtils.getAudioTrackInfo(self._player)
-                let textTracks = await RCTVideoUtils.getTextTrackInfo(self._player)
-                self.onVideoLoad?(["duration": NSNumber(value: duration),
-                                   "currentTime": NSNumber(value: Float(CMTimeGetSeconds(_playerItem.currentTime()))),
-                                   "canPlayReverse": NSNumber(value: _playerItem.canPlayReverse),
-                                   "canPlayFastForward": NSNumber(value: _playerItem.canPlayFastForward),
-                                   "canPlaySlowForward": NSNumber(value: _playerItem.canPlaySlowForward),
-                                   "canPlaySlowReverse": NSNumber(value: _playerItem.canPlaySlowReverse),
-                                   "canStepBackward": NSNumber(value: _playerItem.canStepBackward),
-                                   "canStepForward": NSNumber(value: _playerItem.canStepForward),
-                                   "naturalSize": [
-                                       "width": width,
-                                       "height": height,
-                                       "orientation": orientation,
-                                   ],
-                                   "audioTracks": audioTracks,
-                                   "textTracks": self._textTracks?.compactMap { $0.json } ?? textTracks.map(\.json),
-                                   "target": self.reactTag as Any])
+                self._videoLoadStarted = false
             }
-
+        } else {
             self._videoLoadStarted = false
-            self._playerObserver.attachPlayerEventListeners()
-            self.applyModifiers()
         }
+
+        self._playerObserver.attachPlayerEventListeners()
```

Comment:
Is it really expected to move these events outside of the Task ?

Reply:
I moved whole code into Task